### ISSUE TITLE
Bokeh tool bar disappears when data type inside a DynamicMap changes

### ISSUE THREAD

**Issue by basnijholt** [pyviz/holoviews#2784]
![bokeh](https://user-images.githubusercontent.com/6897215/41138880-2c40690a-6a99-11e8-8238-b15c20deec01.gif)

```python
import holoviews as hv
hv.notebook_extension('bokeh')

def plot():
    while True:
        if path:
            yield hv.Overlay([hv.Path([])])
        else:
            yield hv.Overlay([hv.Scatter([])])

path = True
dm = hv.DynamicMap(plot(), streams=[hv.streams.Stream.define('Next')()])
dm
```
then run
```python
path = False
dm.event() 
```

This currently blocks https://gitlab.kwant-project.org/qt/adaptive/merge_requests/72
Somewhat related https://github.com/ioam/holoviews/issues/2494

**Comment by philippjfr** [comment]
This is a bokeh issue to do with the way renderers are synced between frontend and backend and nothing on our end will fix it. The related bokeh issues are https://github.com/bokeh/bokeh/issues/7497 and https://github.com/bokeh/bokeh/issues/7590. This is precisely why I was worried about adding the ability to dynamically add new subplots, but iirc you persuaded me that we should go ahead and merge that despite this issue.

**Comment by basnijholt** [comment]
> but iirc you persuaded me that we should go ahead and merge that despite this issue

That's true, I remember now! (and I still agree but my past self 😜)

**Comment by philippjfr** [comment]
Anyway, filing an issue is still sensible to track the progress in bokeh.

**Comment by basnijholt** [comment]
It seems like https://github.com/bokeh/bokeh/pull/8085 (PR) fixes https://github.com/bokeh/bokeh/issues/7497 and https://github.com/bokeh/bokeh/issues/7590 is already fixed 😄

**Comment by basnijholt** [comment]
https://github.com/bokeh/bokeh/pull/8085 is finally merged and it seems that it will be in `bokeh` 1.1.0.

**Comment by philippjfr** [comment]
Yes, this will be fixed once we release hv 1.12.0 some time next week. The current release of holoviews will not support bokeh 1.1.0.

**Comment by philippjfr** [comment]
I think we can just merge this since it'll be fixed without any action on our part.
